tests/printer_test.py

```diff
@@ -264,3 +283,128 @@ def test_print_costum_name():
     printer = Printer(stream=file)
     printer.print_op(module)
     assert file.getvalue().strip() == expected.strip()
+
+
+#   ____          _                  _____                          _
+#  / ___|   _ ___| |_ ___  _ __ ___ |  ___|__  _ __ _ __ ___   __ _| |_
+# | |  | | | / __| __/ _ \| '_ ` _ \| |_ / _ \| '__| '_ ` _ \ / _` | __|
+# | |__| |_| \__ \ || (_) | | | | | |  _| (_) | |  | | | | | | (_| | |_
+#  \____\__,_|___/\__\___/|_| |_| |_|_|  \___/|_|  |_| |_| |_|\__,_|\__|
+#
+
+
+@irdl_op_definition
+class PlusCustomFormatOp(Operation):
+    name = "test.add"
+    lhs = OperandDef(IntegerType)
+    rhs = OperandDef(IntegerType)
+    res = ResultDef(IntegerType)
+
+    @classmethod
+    def parse(cls, result_types: List[Attribute],
+              parser: Parser) -> PlusCustomFormatOp:
+        lhs = parser.parse_ssa_value()
+        parser.skip_white_space()
+        parser.parse_char("+")
+        rhs = parser.parse_ssa_value()
+        return PlusCustomFormatOp.create(operands=[lhs, rhs],
+                                         result_types=result_types)
+
+    def print(self, printer: Printer):
```

Comment:
Food for thought: It might be cool, if we could just write something like 
```
printer.print(f"{self.lhs} + {self.rhs}")
```
here, tough I guess that does not work in our current infrastructure as the SSAValue names are not known...

Reply:
Yes that would look better I feel. You mentioned ssomething about refactoring printer and parser at some point, maybe we can make it such that this would be possible afterward?

Reply:
Yeah, though I'm not completely sure that I'll be able to do that :/
The problem is that I don't know how we can print an SSA Value without a context

Reply:
Though what we could do is to write `assembly_format = "$lhs + $rhs"` if you want.
Though we could do this in a separate PR

Reply:
Yeah let's leave this one as is, I think its fine like this.